From: Kir Kolyshkin <kir@openvz.org>
Date: Tue, 31 Mar 2009 12:05:53 +0000 (+0400)
Subject: etc/init.d/vz*: check for DST_CRONSCRIPT_DIR
X-Git-Url: http://git.openvz.org/?p=vzctl;a=commitdiff_plain;h=d4d3bca4052daaf265946159a71fc9ed16c33cb8

etc/init.d/vz*: check for DST_CRONSCRIPT_DIR

Before commits df09d4ed and 00a5a95c we used 'install' to put the cronscript
into DST_CRONSCRIPT_DIR (/etc/cron.d), and it was making sure that DST
dir existed.

Since we're now using 'cat' not 'install' we have to check for DST directory
explicitly. More to say, if there's no /etc/cron.d/ than there's probably
no sense to create it (crond is missing or smth). So just silently exit.

Should fix http://bugzilla.openvz.org/1078

Reported-by: Peter Volkov <pva@gentoo.org>
Signed-off-by: Kir Kolyshkin <kir@openvz.org>
---

diff --git a/etc/init.d/vz-gentoo.in b/etc/init.d/vz-gentoo.in
index b1a1a59..8836768 100755
--- a/etc/init.d/vz-gentoo.in
+++ b/etc/init.d/vz-gentoo.in
@@ -58,6 +58,7 @@ setup_cron()
 {
 	[ -z "$SRC_CRONSCRIPT_DIR" ] && return
 	[ -d "$SRC_CRONSCRIPT_DIR" ] || return
+	[ -d "$DST_CRONSCRIPT_DIR" ] || return
 	cat $SRC_CRONSCRIPT_DIR/vz* > $DST_CRONSCRIPT_DIR/vz &&
 		chmod 644 $DST_CRONSCRIPT_DIR/vz
 }
@@ -66,6 +67,7 @@ remove_cron()
 {
 	[ -z "$SRC_CRONSCRIPT_DIR" ] && return
 	[ -d "$SRC_CRONSCRIPT_DIR" ] || return
+	[ -d "$DST_CRONSCRIPT_DIR" ] || return
 	cat > $DST_CRONSCRIPT_DIR/vz <<EOF
 # DO NOT EDIT THIS FILE!
 #
diff --git a/etc/init.d/vz-redhat.in b/etc/init.d/vz-redhat.in
index a75c2b3..7733389 100755
--- a/etc/init.d/vz-redhat.in
+++ b/etc/init.d/vz-redhat.in
@@ -129,6 +129,7 @@ setup_cron()
 	check_old_cron_files
 	[ -z "$SRC_CRONSCRIPT_DIR" ] && return
 	[ -d "$SRC_CRONSCRIPT_DIR" ] || return
+	[ -d "$DST_CRONSCRIPT_DIR" ] || return
 	cat $SRC_CRONSCRIPT_DIR/vz* > $DST_CRONSCRIPT_DIR/vz &&
 		chmod 644 $DST_CRONSCRIPT_DIR/vz
 }
@@ -138,6 +139,7 @@ remove_cron()
 	check_old_cron_files
 	[ -z "$SRC_CRONSCRIPT_DIR" ] && return
 	[ -d "$SRC_CRONSCRIPT_DIR" ] || return
+	[ -d "$DST_CRONSCRIPT_DIR" ] || return
 	cat > $DST_CRONSCRIPT_DIR/vz <<EOF
 # DO NOT EDIT THIS FILE!
 #
